content/post/20200426-kubernetes-the-universal-control-plane/index.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "Kubernetes: The Universal Control Plane"
+date: 2020-04-26T23:02:29-07:00
+tags:
+  - Kubernetes
+  - cloud
+---
+
+> The innovation is in the API
+
+*This is the first in a 2 part series on Kubernetes, exploring what changes with the project in the picture. Part one is an answer: What are the key ideas that influence the design of Kubernetes? What about Kubernetes sets it apart from other platforms? Part two is about the second order effects: How does the industry respond to a universal control plane? What will our tooling, practices, and platforms look like in the future?*
+
+Ask a developer what Kubernetes is about, and the response would probably something related to container orchestration. It is a fair answer, considering the description on the [homepage](https://kubernetes.io/):
+
+> Kubernetes (K8s) is an open-source system for automating deployment, scaling, and management of containerized applications.
+
+During the fierce container wars of the past few years, Kubernetes positioned itself as the dominant container platform. But in the long term I think the vision of Kubernetes centers not around containers, but its API, and it attempts to be a platform for managing software at a more fundamental and encompassing level.
+
+To put it briefly, I think the goal is for the Kubernetes API to be the **universal control plane of software**. The API aims to be the authoritative interface for managing software. If we can model the domain as a resource, Kubernetes should manage it. Successful execution of the project's strategy will shift cloud computing, software infrastructure, and Kubernetes itself.
+
+## Looking Back: Principles of Cloud Computing
+
+Before we get into control planes, we need to take a step back and analyze the rise and rapid adoption of cloud computing for running software workloads. A straightforward reason to do so is because Kubernetes was influenced by and now influences the cloud computing industry. But at a more subtle level, building Infrastructure as a Service (IaaS) requires techniques to make the endeavour operable and practical at scale. Here are two insights from building cloud infrastructure:
```

Comment:
you don't need "for running software workloads" that's just what computing does.

Next sentence I would remove "straightforward" with "obvious" since it feels a little less wordy and I would merge it with the next sentence. Instead of cutting at But just use a comma. About that next sentence, I'm not sure I see a good cohesion between what you're saying so might want to clarify why these "techniques" are relevant to understanding k8. I would also remove the word "operable" and just use practical since they are basically saying the same thing here.